modules/porous_flow/include/actions/PorousFlowSinglePhaseBase.h

```diff
@@ -0,0 +1,57 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef POROUSFLOWSINGLEPHASEBASE_H
+#define POROUSFLOWSINGLEPHASEBASE_H
+
+#include "PorousFlowActionBase.h"
+
+class PorousFlowSinglePhaseBase;
+
+template <>
+InputParameters validParams<PorousFlowSinglePhaseBase>();
+
+/**
+ * Base class for actions involving a single fluid phase
+ */
+class PorousFlowSinglePhaseBase : public PorousFlowActionBase
+{
+public:
+  PorousFlowSinglePhaseBase(const InputParameters & params);
+
+  virtual void act() override;
+
+protected:
+  virtual void addDictator() override;
+
+  /// porepressure NonlinearVariable name
+  const NonlinearVariableName _pp_var;
+
+  /// Determines the coupling type
+  const enum CouplingTypeEnum {
+    Hydro,
+    ThermoHydro,
+    HydroMechanical,
+    ThermoHydroMechanical
+  } _coupling_type;
+
+  /// whether steady or transient simulation
+  const enum SimulationTypeChoiceEnum { STEADY, TRANSIENT } _simulation_type;
+
+  /// Name of the fluid-properties UserObject
+  const UserObjectName _fp;
```

Comment:
This should be a reference. I'm not sure how this isn't breaking right now. Are you sure the information is getting updated?